www/nginx/src/opnsense/mvc/app/controllers/OPNsense/Nginx/Api/SettingsController.php

```diff
@@ -342,7 +342,69 @@ public function sethttprewriteAction($uuid)
     // http security headers
     public function searchsecurityHeaderAction()
     {
-        return $this->searchBase('security_header', array('description'));
+        $data = $this->searchBase('security_header',
+            ['description', 'referrer', 'xssprotection', 'strict_transport_security_time', 'hpkp_keys', 'hpkp_time',
+            'enable_csp', 'csp_report_only', 'csp_default_src_enabled', 'csp_script_src_enabled', 'csp_img_src_enabled',
+            'csp_style_src_enabled', 'csp_media_src_enabled', 'csp_font_src_enabled', 'csp_frame_src_enabled',
+            'csp_frame_ancestors_enabled', 'csp_form_action_enabled']);
+
+        // Create "hsts" column (disabled/time)
+        foreach ($data['rows'] as &$row) {
+            if (intval($row['strict_transport_security_time']) > 0) {
+                $row['hsts'] = sprintf("%d %s", intval($row['strict_transport_security_time']), gettext("sec"));
+            } else {
+                $row['hsts'] = gettext('disabled');
+            }
+        }
+
+
+        // Create "hpkp" column (enabled/disabled)
+        foreach ($data['rows'] as &$row) {
+            if (!empty($row['hpkp_keys']) && intval($row['hpkp_time']) > 0) {
+                $row['hpkp'] = sprintf("%d %s", intval($row['hpkp_time']), gettext("sec"));
+            } else {
+                $row['hpkp'] = gettext('disabled');
+            }
+        }
```

Comment:
Ich denke, HPKP kann weg, wird nicht mehr unterstützt:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Public-Key-Pins

Der Ersatz heißt anscheinend Expect-CT:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect-CT

Reply:
Guter Punkt... Denkst du sollen wir es gänzlich entfernen oder im UI "depracated" incl. Erklärung ergänzen?

Reply:
In anbetracht dessen, dass es nur einen Effekt in sehr alten Browsern hätte (EoL), könnten wir es bedenkenlos entfernen.

Ich denke, der Grund dafür ist, dass HPKP sehr oft zum lockout geführt hat, Wenn der User zur Zeit des Key-Rollover mal die Seite nicht angesurft hat.

Reply:
> Der Ersatz heißt anscheinend Expect-CT:
> 
> https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect-CT

Auch das scheint schon obsolet zu sein:

_Note: The Expect-CT will likely become obsolete in June 2021. Since May 2018 new certificates are expected to support SCTs by default. Certificates before March 2018 were allowed to have a lifetime of 39 months, those will all be expired in June 2021._

Ich schlage vor nun HKPK zu entfernen und den Nachfolger separat zu behandeln.